libmscore/keysig.cpp

```diff
@@ -101,7 +101,7 @@ void KeySig::layout()
             }
 
       _sig.keySymbols().clear();
-      if (staff() && !staff()->genKeySig())     // no key sigs on TAB staves
+      if (staff() && !staff()->staffType(tick())->genKeysig())     // no key sigs on TAB staves
```

Comment:
This causes tablature staves to display key signatures which doesn't make sense for them. I would rather suggest to generalize the existing [`Staff::genKeySig()`](https://github.com/musescore/MuseScore/blob/ee4196a018fb7ae0421260fdbe845aed5a5891b4/libmscore/staff.cpp#L1131) function to work for arbitrary tick rather than tick 0 and use it here instead, then tablature staves would be handled correctly.

Reply:
What about changing `staffType(tick())->genKeysig()` to handle TAB staves correctly? Because it isn't very consistent if `staff()->gensth()` is used here while `staffType(tick())->gensth()` is used everywhere else.

Reply:
Yes, I believe that should also work well.